src/elements/src/index.ts

```diff
@@ -0,0 +1,55 @@
+import { LifeCycles } from 'single-spa';
+import { NgElement } from '@angular/elements';
+
+import {
+  SingleSpaAngularElementsOptions,
+  BootstrappedSingleSpaAngularElementsOptions,
+} from './types';
+
+const defaultOptions: BootstrappedSingleSpaAngularElementsOptions = {
+  element: null!,
+  ngModuleRef: null,
+  bootstrapFunction: null!,
+  withProperties: undefined,
+  domElementGetter: undefined,
+};
+
+async function bootstrap(options: BootstrappedSingleSpaAngularElementsOptions) {
+  if (options.ngModuleRef !== null) {
+    return;
+  }
+
+  options.ngModuleRef = await options.bootstrapFunction();
+}
+
+async function mount(options: BootstrappedSingleSpaAngularElementsOptions) {
+  if (typeof options.withProperties === 'function') {
+    const properties = await options.withProperties(options.ngModuleRef!);
+    const element = document.querySelector(options.element) as NgElement;
+
+    for (const [property, value] of Object.entries(properties)) {
+      element[property] = value;
+    }
+  }
+}
+
+function unmount(options: BootstrappedSingleSpaAngularElementsOptions) {
+  const node = document.querySelector(options.element);
+  // Removing custom element from DOM is enough since it will trigger
+  // `disconnectedCallback()` and Angular will dispose all resources.
+  node?.parentElement?.removeChild(node);
```

Comment:
My guess is that you added the `?.` to make typescript happy. However, we want it to throw an error when `node` is null. If we're selecting the wrong element, we want the lifecycle function to reject its promise so that the single-spa application/parcel moves into SKIP_BECAUSE_BROKEN status.

Idk if typescript is smart enough to infer types from if/else statements, but if so perhaps the following could make it happy?

```js
if (node) {
  node.parentElement.removeChild()
} else {
  throw Error(`Could not find Angular element with selector ${options.element}`)
}
```